src/VirtoCommerce.CatalogExportImportModule.Data/Validation/ImportPhysicalProductValidator.cs

```diff
@@ -176,6 +176,44 @@ private void AttachValidators()
                 .WithInvalidValueCodeAndMessage("Tax Type")
                 .WithImportState();
 
+            // Variations
+            RuleFor(record => record.Record.MainProductId)
+                .Must((_, mainProductId, context) =>
+                {
+                    var existedMainProducts =
+                        (CatalogProduct[])context.ParentContext.RootContextData[
+                            ModuleConstants.ValidationContextData.ExistedMainProducts];
+
+                    var result = existedMainProducts.Any(x => x.Id.EqualsInvariant(mainProductId));
+
+                    return result;
+                })
+                .WithNotExistedMainProduct()
+                .WithImportState()
+                .Must((record, mainProductId, _) =>
+                {
+                    var result = !mainProductId.EqualsInvariant(record.Record.ProductId);
+
+                    return result;
+                })
+                .WithSelfCycleReference()
+                .WithImportState()
+                .Must((_, mainProductId, context) =>
+                {
+                    var existedMainProducts =
+                        (CatalogProduct[])context.ParentContext.RootContextData[
+                            ModuleConstants.ValidationContextData.ExistedMainProducts];
+
+                    var mainProduct = existedMainProducts.FirstOrDefault(x => x.Id.EqualsInvariant(mainProductId));
+
+                    var mainProductIsNotVariation = string.IsNullOrEmpty(mainProduct?.MainProductId);
+
+                    return mainProductIsNotVariation;
+                })
+                .WithMainProductIsVariation()
+                .WithImportState()
+                .When(record => !string.IsNullOrEmpty(record.Record.MainProductId));
```

Comment:
What about the same checks, but for `Main Product Outer Id`? What if they are for different products?